utils/terrascript_client.py

```diff
@@ -435,17 +436,126 @@ def populate_tf_resource_s3(self, resource, namespace_info):
         # https://www.terraform.io/docs/providers/aws/r/s3_bucket.html
         values = {}
         values['bucket'] = identifier
-        values['versioning'] = {'enabled': True}
+        values['versioning'] = common_values.get('versioning') or False
         values['tags'] = common_values['tags']
         values['acl'] = common_values.get('acl') or 'private'
         values['server_side_encryption_configuration'] = \
             common_values.get('server_side_encryption_configuration')
         if common_values.get('lifecycle_rules'):
             # common_values['lifecycle_rules'] is a list of lifecycle_rules
             values['lifecycle_rule'] = common_values['lifecycle_rules']
+        sc = common_values.get('storage_class')
+        if sc:
+            rule = {
+                "id": sc + "_storage_class",
+                "enabled": "true",
+                "transition": {
+                    "storage_class": sc
+                }
+            }
+            if values.get('lifecycle_rule'):
+                values['lifecycle_rule'].append(rule)
+            else:
+                values['lifecycle_rule'] = rule
         if common_values.get('cors_rules'):
             # common_values['cors_rules'] is a list of cors_rules
             values['cors_rule'] = common_values['cors_rules']
+        if common_values.get('replication_configurations'):
+            rc_configs = []
+            for config in common_values['replication_configurations']:
```

Comment:
can we reduce to a single access to the dict? just to be consistent with the above:
```
sc = common_values.get('storage_class')
        if sc:
```

Reply:
Done